### PR TITLE
Fix Clang deprecated builtins

### PR DESCRIPTION
**Pull request by adamscott**
It seems that Clang and GCC have different interpretations of certain builtins. So this PR fixes the "deprecated" builtins for Clang, but keeps them for GCC.

Fixes https://github.com/godotengine/godot-cpp/issues/1210

### CODE REVIEW

File: include/godot_cpp/templates/cowdata.hpp [hunk 1]
```diff
@@ -321,7 +333,11 @@ Error CowData<T>::resize(int p_size) {
 		*_get_size() = p_size;
 
 	} else if (p_size < current_size) {
+#ifdef __clang__
+		if (!__is_trivially_constructible(T)) {
```
Comment (dsnopek) [inline include/godot_cpp/templates/cowdata.hpp:337]: To match how this was converted above, should this be `__is_trivially_destructible(T)`?

Comment (adamscott) [reply include/godot_cpp/templates/cowdata.hpp]: Thanks! I updated the code to fix it.

**Comment by dsnopek** [comment]
Thanks!

Looking at `cowdata.h` in Godot, it's using the functions from the C++11, ie `std::is_trivially_destrucible()`, etc. Perhaps it would be better to sync up with Godot, rather than adding the `#ifdef`s?

**Comment by adamscott** [comment]
> Looking at `cowdata.h` in Godot, it's using the functions from the C++11, ie `std::is_trivially_destrucible()`, etc. Perhaps it would be better to sync up with Godot, rather than adding the `#ifdef`s?

Done!

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**